### ISSUE TITLE
BUG: CI on Azure broken with PyTest 7.1

### ISSUE THREAD

**Issue by tupui** [scipy/scipy#15780]
Seems like the new release of PyTest broke the CI on Azure. Ex. https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=16988&view=logs&jobId=7d84eb14-5b07-5555-8e62-838a7038a2b0&j=ede89041-5e8e-5201-ce5c-f68c2cbf02dd&t=141d9d53-a127-58b4-0262-4b9693c8e91f

```
E   pytest.PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
```

https://docs.pytest.org/en/latest/changelog.html

I am not 100% sure it's due to Pytest, but the timing seems to corresponds.

**Comment by ilayn** [comment]
Yes this was not the case with 7.0.1 but now with 7.1.0 on Azure workflows mark.slow errors out

**Comment by tupui** [comment]
I did not see any other update of dependencies nor don't see anything we changed recently that could be causing this. But locally I could not reproduce, so not sure.

**Comment by tupui** [comment]
Adding a milestone to not forget to check this.

**Comment by ilayn** [comment]
Pytest folks detected the problem and relevant change will be soon I presume.

**Comment by tupui** [comment]
So they fixed it and released 7.1.1. I am not sure what to do between `pytest != 7.1.0, pytest >=7.1.1` or just `pytest`. In the end there is little change a user would get bitten by that and our CI would just pick latest.
